### PR TITLE
Add count_include_pad arg for average_pool_op on CPU

### PR DESCRIPTION
**Pull request by xiaomengy**
Summary: Add count_include_pad arg for average_pool_op on CPU

Differential Revision: D13558123

### CODE REVIEW

File: caffe2/operators/pool_op.cc [hunk 1]
```diff
@@ -900,22 +536,20 @@ std::function<void(OpSchema&)> MaxPoolDocGenerator(const char* dim) {
...
+    string; default: 'NCHW')* Order of the blob dimensions.");
     */
   };
 }
 REGISTER_CPU_OPERATOR(
     AveragePool,
-    PoolOp<float, CPUContext, AveragePool<float>>);
+    PoolOp<float, CPUContext, AveragePoolFunctor<CPUContext>>);
 
 OPERATOR_SCHEMA(AveragePool)
```
Comment (houseroad) [inline caffe2/operators/pool_op.cc]: Shall we also add the description about arg `count_include_pad` to the AveragePoolDocGenerator?

File: caffe2/operators/pool_op_util.cc [hunk 2]
```diff
@@ -0,0 +1,922 @@
...
+  (void)H;
+  (void)W;
+  (void)X;
+  (void)Y;
+#endif
+}
+
+#define CAFFE2_SPECIALIZED_RUN_AVERAGE_POOL_1D(T, kCountIncludePad)    \
+  template <>                                                          \
+  void RunAveragePool1D<T, StorageOrder::NCHW, kCountIncludePad>(      \
```
Comment (houseroad) [inline caffe2/operators/pool_op_util.cc]: Can we add kCountIncludePad as argument?

**Review by houseroad — COMMENTED** [review]
Could you also add a test case to cover `count_include_pad` argument?

File: caffe2/operators/pool_op.cc (hunk 1 above)
Comment (BIT-silence) [reply caffe2/operators/pool_op.cc]: Actually I separate this approach into two parts. This is the first part to add it into CPU. And then I will add it into GPU part, which is more important. So I'd suggest add the doc when the GPU part finished.

File: caffe2/operators/pool_op_util.cc (hunk 2 above)
Comment (BIT-silence) [reply caffe2/operators/pool_op_util.cc]: It is trying to make the design of the related functions on CPU and GPU the same. On GPU part, if there is a condition between different threads in runtime, there will be some negative impact on the performance.

**Review by houseroad — APPROVED** [review]
Discussed offline, the comments will be addressed in the following PRs.

**Pull request closed without merging**